packages/components/src/internal/components/permissions/GroupDetailsPanel.tsx

```diff
@@ -2,38 +2,122 @@
  * Copyright (c) 2018-2019 LabKey Corporation. All rights reserved. No portion of this work may be reproduced in
  * any form or by any electronic or mechanical means without written permission from LabKey Corporation.
  */
-import React from 'react';
-import { Panel } from 'react-bootstrap';
+import React, { FC, memo, useCallback, useEffect, useMemo, useState } from 'react';
+import { Col, Panel, Row } from 'react-bootstrap';
 import { Map } from 'immutable';
 
-import { Principal, SecurityPolicy, SecurityRole } from './models';
+import { Filter, Query } from '@labkey/api';
+
+import { resolveErrorMessage } from '../../util/messaging';
+import { Member } from '../administration/models';
+
 import { EffectiveRolesList } from './EffectiveRolesList';
 
+import { Principal, SecurityPolicy, SecurityRole } from './models';
+
+function getWhenCreated(id: number): Promise<string> {
```

Comment:
this function looks very similar to the `getLastModified` function in the `GroupManagement.tsx` file. Could they be shared/consolidated in some way?

Reply:
Also, should they maybe be using the `selectRows` method in this repo?